### PR TITLE
DOC: known issue about pandas Series multiplication

### PR DESCRIPTION
**Pull request by MridulS**
### Description

This PR adds a section to https://docs.astropy.org/en/latest/known_issues.html regarding the fragile behavior with pd.Series objects.

Fixes [#11247](https://github.com/astropy/astropy/issues/11247)

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by MridulS** [comment]
Hmm, this is failing as pandas isn't installed in the minimal dependencies environment and in the `olddeps` environment it fails with pandas 1.4.

File: docs/known_issues.rst
Comment (mhvk) [inline docs/known_issues.rst:103]: Maybe add: 
```
What should be secure, though, is to initialize the |Quantity| directly::
    >>> u.Quantity(a, u.m)
    <Quantity [1., 2., 3.] m>
```

**Review by mhvk — COMMENTED** [review]
On the test failure, there is 
```
.. doctest-requires:: pandas
```
but I'm not sure we can give this a minimum version.

To be honest, I'm fine with just skipping the tests...

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:79]: This might be true, but there are no examples about that. A title that would actually match the text would be
```suggestion
Multiplying a `pandas.Series` with an `~astropy.units.Unit` does not produce a |Quantity|
-----------------------------------------------------------------------------------------
```

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:83]: `Quantity` is indeed a subclass of `ndarray`, but that's not relevant here.

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:91]: Do run the tests is possible. If the test fails with `pandas` 1.4 then use
```suggestion
.. doctest-requires:: pandas>=1.5

   >>> import pandas as pd # doctest: +SKIP
   >>> import astropy.units as u # doctest: +SKIP
   >>> a = pd.Series([1., 2., 3.]) # doctest: +SKIP
   >>> a * u.m # doctest: +SKIP
```
Also, if different versions of `pandas` produce different outcomes then it's important to document that, so we might need multiple sets of tests.

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:100]: The text mentions in-place left shift operator (`<<=`), but the example uses the left shift operator (`<<`). It would be best to demonstrate both.

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:103]: If we run doctests here with the development or latest version of `pandas` then we don't need to worry about this happening without us noticing.

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:104]:
```suggestion
secure, though, is to initialize the |Quantity| directly::
```

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:86]:
```suggestion
For example, multiplying a `~pandas.Series` instance
with a unit will *not* return a |Quantity|. It will return a `~pandas.Series`
object without any unit::
```

File: docs/known_issues.rst
Comment (MridulS) [reply docs/known_issues.rst:91]: Pandas 1.4 should be dropped according to [SPEC 0](https://scientific-python.org/specs/spec-0000/), I'm not sure if we should write specific docs for that? `olddeps` should probably drop pandas 1.4 soon?

File: docs/known_issues.rst
Comment (MridulS) [reply docs/known_issues.rst:103]: True, but I guess it's still nice to document the "right" way to initialize a quantity with a pandas object?

Comment (eerovaher) [reply docs/known_issues.rst:109]: I don't really know what to think of this sentence. By providing the example we are recommending using the shift operators. But the text is saying they are fragile and it's best not to use them. We shouldn't be making mixed statements like this.

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:114]: The previous example converted `a` to a `Quantity`, so `pandas` is not involved here at all.

File: docs/known_issues.rst
Comment (eerovaher) [reply docs/known_issues.rst]: Dropping support for `pandas` 1.4 sounds like a good idea, but the pull request doing that should be merged first.

File: docs/known_issues.rst
Comment (mhvk) [reply docs/known_issues.rst:109]: Hmm, @eerovaher, good point. I think it may still be OK to list the operator, but let's put `u.Quantity` first. I.e., something like,
```
To avoid this, best is just to initialize as a `Quantity`,

   >>> u.Quantity(a, u.m)
...
Note that the overrides pandas provides are not complete, and
as a consequence, using the (in-place) shift operator does work,
    >>> a << u.m
...
But this ...
```

File: docs/known_issues.rst
Comment (MridulS) [reply docs/known_issues.rst]: Sounds good, created https://github.com/astropy/astropy/pull/16023

File: docs/known_issues.rst
Comment (MridulS) [reply docs/known_issues.rst:109]: I like @mhvk suggestion. The reason I do think it makes sense to keep some docs about this fragile behavior is it seems like users are trying to use this today and they are surprised it works for some operations and it doesn't for some. The fragile behavior seems to stems from the user assumption that numpy array and pandas series should work similarly (?). 

Yes we can tell them to just use `u.Quantity(a, u.m)` directly but maybe it's nice to also explain why their expectation (Series ~ ndarray) doesn't work?

**Review by mhvk — APPROVED** [review]
Thanks, @MridulS! Looks all good to me, but I'll let @eerovaher have a last look too.

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:111]: Would this be clearer ?
```suggestion
   >>> b = a << u.m
   >>> b
   <Quantity [1., 2., 3.] m>
```

**Comment by eerovaher** [comment]
This is almost ready, but there's a trailing whitespace somewhere that is causing the `pre-commit.ci` job to fail.

**Comment by pllim** [comment]
pre-commit.ci autofix

**Comment by pllim** [comment]
@eerovaher , the bot took care of the trailing whitespace. Does this look good to you now? We can always use the squash and merge button for this PR.

Thanks, all!

File: docs/known_issues.rst
Comment (eerovaher) [inline docs/known_issues.rst:87]: The `doctest-requires` instructions are appearing in the rendered documentation.
```suggestion
object without any unit:

.. doctest-requires:: pandas>=1.5
```
I'm only highlighting it here, but the others need to be edited too.

**Review by eerovaher — APPROVED** [review]

**Pull request merged**
